conversion.go

```diff
@@ -0,0 +1,129 @@
+package safecast
+
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+type Integer interface {
+	~int | ~uint | ~int8 | ~uint8 | ~int16 | ~uint16 | ~int32 | ~uint32 | ~int64 | ~uint64
+}
+
+var ErrOutOfRange = errors.New("out of range")
+
+func ToInt[T Integer](i T) (int, error) {
+	if i > 0 && uint64(i) > math.MaxInt {
+		return 0, fmt.Errorf("%w: %d is greater than math.MaxInt", ErrOutOfRange, i)
+	}
+
+	if i < 0 && uint64(-i) > -math.MinInt {
```

Comment:
that one looks fishy/odd why not use int64 which is the biggest anyway that can hold a negative number?